src/ops/logical.rs

```diff
@@ -0,0 +1,554 @@
+use crate::array::Array;
+use crate::stream::StreamOrDevice;
+use mlx_macros::default_device;
+
+impl Array {
+    /// Unary element-wise logical not.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use mlx::Array;
+    /// let a: Array = false.into();
+    /// let mut b = a.logical_not_device(Default::default());
+    ///
+    /// b.eval();
+    /// let b_data: &[bool] = b.as_slice();
+    /// // b_data == [true]
```

Comment:
Maybe do `assert_eq!()` here so that cargo test also checks for correctness?

Reply:
We decided not to do this after investigation